src/main/repository/PromotionRepository.ts

```diff
@@ -216,6 +224,43 @@ export class PromotionRepository extends Repository<Promotion> {
       return promotion;
     });
   }
+
+  /**
+   * Like as we did in findPromotionUserSaved, check all promotions is voted by the user
+   * @param userId the id of the user
+   * @param promotions the promotions we want to find out if the user voted
+   */
+  private async findPromotionsUserVoted(
+    userId: string,
+    promotions: Promotion[]
+  ): Promise<Promotion[]> {
+    const promotionIds = promotions.map((promotion: Promotion) => promotion.id);
+    const voteRecordIds = await getConnection()
+      .createQueryBuilder(VoteRecord, 'voteRecords')
+      .select(['voteRecords.promotionId', 'voteRecords.voteState'])
+      .where('voteRecords.userId = :userId', { userId })
+      .andWhere('voteRecords.promotionId IN (:...promotionIds)', {
+        promotionIds,
+      })
+      .getRawMany();
```

Comment:
Ah sorry I made changes in my other PR for `findPromotionUserSaved`. 

Try following this, the benefit is that it will return an array of VoteRecord, so we can leverage type safety at compile time
https://github.com/ubclaunchpad/PromoPal-backend/pull/131/files#diff-45ba25f3f1fe17fcdfb9215942531a829664bc5de9e9bec00c3e08a1e25ff66aL203-R212

Reply:
Oh, ok. I'll refer it!